epochStart/bootstrap/epochStartMetaBlockProcessor.go

```diff
@@ -318,25 +308,47 @@ func (e *epochStartMetaBlockProcessor) requestMetaBlock() error {
 	return nil
 }
 
-func (e *epochStartMetaBlockProcessor) requestConfirmationMetaBlock(nonce uint64) error {
+func (e *epochStartMetaBlockProcessor) requestProofForMetaBlock(metablockHash []byte) error {
 	numConnectedPeers := len(e.messenger.ConnectedPeers())
-	err := e.requestHandler.SetNumPeersToQuery(factory.MetachainBlocksTopic, numConnectedPeers, numConnectedPeers)
+	topic := common.EquivalentProofsTopic + core.CommunicationIdentifierBetweenShards(core.MetachainShardId, core.AllShardId)
+	err := e.requestHandler.SetNumPeersToQuery(topic, numConnectedPeers, numConnectedPeers)
 	if err != nil {
 		return err
 	}
 
-	e.requestHandler.RequestMetaHeaderByNonce(nonce)
+	e.requestHandler.RequestEquivalentProofByHash(core.MetachainShardId, metablockHash)
 
 	return nil
 }
 
+func (e *epochStartMetaBlockProcessor) receivedProof(proof data.HeaderProofHandler) {
+	startOfEpochMetaBlock, hash, err := e.getMostReceivedMetaBlock()
```

Comment:
should we handle received proof only if epoch start meta block has reached the peers list target? i'm thinking on the scenario when we are still `waitingForMetaBlock` and in the meantime a proof is received and we are checking the most currently received meta block

Reply:
compare maybe proof header hash with `e.metaBlockHash` since it will be set if peers list target is reached? this callback will be called on every received proof and it might be better to exit earlier with a simpler check (most of the cases hashes will not match)

Reply:
updated to check against both of them and use the proper one that matches the proof header